[Skia] Implement FontPlatformData::openTypeTable

TingPing opened the pull request:
#### 38b6c344ecebfc290a84fb05f7bd20daeb24c618
<pre>
[Skia] Implement FontPlatformData::openTypeTable
<a href="https://bugs.webkit.org/show_bug.cgi?id=270301">https://bugs.webkit.org/show_bug.cgi?id=270301</a>

Reviewed by Carlos Garcia Campos.

Compared against freetype&apos;s implementation.

* Source/WebCore/platform/graphics/skia/FontPlatformDataSkia.cpp:
(WebCore::FontPlatformData::openTypeTable const):

Canonical link: <a href="https://commits.webkit.org/275578@main">https://commits.webkit.org/275578@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a43eb471241ceb2887b0ca308a098987bfdc2804

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/42178 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/21196 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/44572 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/44773 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/38294 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/44485 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/24419 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/18528 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/34936 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/42752 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/18153 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/36325 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/15872 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/15816 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/37360 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/46216 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/38394 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/37688 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/41593 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/16994 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/13980 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/40168 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/18613 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/9453 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/18675 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/18258 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

donny-dont commented on Source/WebCore/platform/graphics/skia/FontPlatformDataSkia.cpp:
```diff
@@ -80,9 +80,20 @@ String FontPlatformData::familyName() const
+        return nullptr;
+
+    SkFontTableTag tag = SkSetFourByteTag((table & 0xff), (table & 0xff00) >> 8, (table & 0xff0000) >> 16, table >> 24);
+    size_t tableSize = typeface->getTableSize(tag);
```
Any reason why we're doing the endianness conversion instead of just using `SkSetFourByteTag` instead of `OT_MAKE_TAG` for `USE(SKIA)`?

TingPing replied:
I switched it over to `OT_MAKE_TAG` but I don't think it matters really.

donny-dont reviewed (commented):
Question before landing to potentially avoid all the conversions.

TingPing committed:
[Skia] Implement FontPlatformData::openTypeTable
https://bugs.webkit.org/show_bug.cgi?id=270301

Reviewed by Carlos Garcia Campos.

Compared against freetype's implementation.

* Source/WebCore/platform/graphics/skia/FontPlatformDataSkia.cpp:
(WebCore::FontPlatformData::openTypeTable const):

Canonical link: https://commits.webkit.org/275578@main
